unit/util/string_utils/join_string.cpp

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************\
+
+Module: Unit tests of join_string
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+/// \file
+/// join_string Unit Tests
+
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <testing-utils/use_catch.h>
+#include <util/string_utils.h>
+
+TEST_CASE(
+  "join_strings() should apply the function argument its passed to the "
+  "elements of the container")
+{
+  std::vector<int> vec{1, 2, 3};
+  auto result =
+    join_strings(std::ostringstream(), vec.begin(), vec.end(), "-", [](int x) {
+      return std::to_string(x);
```

Comment:
Maybe we can have a transformation that actually distinguishes this from the identity? For example, `return x + 1`;?